File: packages/contracts/typescript/src/enums.ts
/**
 * Shared enumerations for the migration platform.
 * Values must match the Python StrEnum values exactly.
 */

/** Discriminator for project vs single-flow input mode. */
export type InputMode = "project" | "single_flow";

/** Severity levels for warnings, gaps, and validation issues. */
export type Severity = "info" | "warning" | "error" | "critical";

/** Categories for migration gaps. */
export type GapCategory =
  | "unsupported_construct"
  | "unresolvable_reference"
  | "partial_support"
  | "connector_mismatch"
  | "dataweave_complexity";

/** Categories for MuleSoft constructs (from spec §7). */
export type ConstructCategory =
  | "trigger"
  | "router"
  | "connector"
  | "error_handler"
  | "transform"
  | "scope"
  | "flow_control";

File: packages/contracts/src/m2la_contracts/enums.py
"""Shared enumerations for the migration platform."""

from enum import StrEnum


class InputMode(StrEnum):
    """Discriminator for project vs single-flow input mode."""

    PROJECT = "project"
    SINGLE_FLOW = "single_flow"


class Severity(StrEnum):
    """Severity levels for warnings, gaps, and validation issues."""

    INFO = "info"
    WARNING = "warning"
    ERROR = "error"
    CRITICAL = "critical"


class GapCategory(StrEnum):
    """Categories for migration gaps."""

    UNSUPPORTED_CONSTRUCT = "unsupported_construct"
    UNRESOLVABLE_REFERENCE = "unresolvable_reference"
    PARTIAL_SUPPORT = "partial_support"
    CONNECTOR_MISMATCH = "connector_mismatch"
    DATAWEAVE_COMPLEXITY = "dataweave_complexity"


class ConstructCategory(StrEnum):
    """Categories for MuleSoft constructs (from spec §7)."""

    TRIGGER = "trigger"
    ROUTER = "router"
    CONNECTOR = "connector"
    ERROR_HANDLER = "error_handler"
    TRANSFORM = "transform"
    SCOPE = "scope"
    FLOW_CONTROL = "flow_control"


class ValidationCategory(StrEnum):
    """Categories for validation checks."""

    MULE_INPUT = "mule_input"
    IR_INTEGRITY = "ir_integrity"
    OUTPUT_INTEGRITY = "output_integrity"
    CONNECTOR_PREFERENCE = "connector_preference"
